### PR TITLE
[flutter_tools] split start app android tests into new file

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Does not refactor globals, but does switch to FakeProcessManager and removes several mocks.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+      command: <String>['adb', '-s', '1234', 'shell', 'am', 'force-stop', 'FlutterApp'],
+    ));
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'pm', 'list', 'packages', 'FlutterApp'],
+    ));
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+    // TODO(jonahwilliams): investigate why this doesn't work.
+    // This doesn't work with the current Android log reader implementation.
+    processManager.addCommand(const FakeCommand(
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: I can't get the observatory discovery to work for this case. Seems like the logs are getting dropped, even if I force the process to not exit immediately.

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        'android.intent.action.RUN',
+        '-f',
+        '0x20000000',
+        // The DebuggingOptions arguments go here.
+        '--ez', 'enable-background-compilation', 'true',
+        '--ez', 'enable-dart-profiling', 'true',
+        '--ez', 'enable-software-rendering', 'true',
+        '--ez', 'skia-deterministic-rendering', 'true',
+        '--ez', 'trace-skia', 'true',
+        '--ez', 'trace-whitelist', 'bar,baz',
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: FYI @gaaclarke

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+      processManager.addCommand(kAdbVersionCommand);
+      processManager.addCommand(kStartServer);
+
+      // This configures the target platform of the device.
+      processManager.addCommand(FakeCommand(
+        command: const <String>['adb', '-s', '1234', 'shell', 'getprop'],
+        stdout: '[ro.product.cpu.abi]: [$arch]'
+      ));
+      processManager.addCommand(const FakeCommand(
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: I could pull a few more of these into consts

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        command: <String>['adb', '-s', '1234', 'shell', 'am', 'force-stop', 'FlutterApp'],
+      ));
+      processManager.addCommand(const FakeCommand(
+        command: <String>['adb', '-s', '1234', 'shell', 'pm', 'list', 'packages', 'FlutterApp'],
+      ));
+      processManager.addCommand(kAdbVersionCommand);
+      processManager.addCommand(kStartServer);
+      processManager.addCommand(const FakeCommand(
+        command: <String>['adb', '-s', '1234', 'install', '-t', '-r', 'app.apk']
```
Comment (blasten) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: nit: trailing comma

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+      versionCode: 1,
+    );
+
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+
+    // This configures the target platform of the device.
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'getprop'],
+      stdout: '[ro.product.cpu.abi]: [x86]'
```
Comment (blasten) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: nit: trailing comma

File: packages/flutter_tools/test/general.shard/android/android_device_start_test.dart [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+      file: apkFile,
+      launchActivity: 'FlutterActivity',
+      versionCode: 1,
+    );
+
+    // These commands are required to install and start the app
+    processManager.addCommand(kAdbVersionCommand);
+    processManager.addCommand(kStartServer);
+    processManager.addCommand(const FakeCommand(
+      command: <String>['adb', '-s', '1234', 'shell', 'getprop']
```
Comment (blasten) [inline packages/flutter_tools/test/general.shard/android/android_device_start_test.dart]: nit: trailing comma

**Review by blasten — APPROVED** [review]
LGTM + minor nits

**Pull request merged**